pkg/apiserver/registry/core/shoot/strategy.go

```diff
@@ -75,8 +91,29 @@ func (shootStrategy) PrepareForUpdate(_ context.Context, obj, old runtime.Object
 		newShoot.Generation = oldShoot.Generation + 1
 	}
 
-	if oldShoot.Spec.CloudProfile == nil && !utilfeature.DefaultFeatureGate.Enabled(features.UseNamespacedCloudProfile) {
-		newShoot.Spec.CloudProfile = nil
+	if newShoot.Spec.CloudProfile == nil && newShoot.Spec.CloudProfileName == nil {
+		newShoot.Spec.CloudProfile = oldShoot.Spec.CloudProfile
+		newShoot.Spec.CloudProfileName = oldShoot.Spec.CloudProfileName
+	}
```

Comment:
Why do we revert this change silently and not let the update failed instead? Today, the immutability and not-empty validation will be forbid this change.

Reply:
Yes, will revert this change. Makes more sense.